src/stores/courses.js

```diff
@@ -1,40 +1,36 @@
-import { reactive } from 'vue';
 import { defineStore } from 'pinia';
-import courseData from '../assets/data/courses.json';
+import { reactive } from 'vue';
 
-export const useCoursesStore = defineStore('courses', () => {
-	let courses = reactive(courseData[0].courses);
+export const useCoursesStore = (courseData, courseId) => {
+  return defineStore('course', () => {
+    let courses = reactive(courseData.courses);
 
-	function getCourseById(id) {
-		return courses.find((course) => course.id == parseInt(id));
-	}
+    const course = courses.find((course) => course.id == courseId);
```

Comment:
Everything in this API is contingent upon there being a course, so just define it upfront and remove the need for callsites to call `getCourseById`.